src/expr/proof_rule.h

```diff
@@ -617,6 +617,15 @@ enum class PfRule : uint32_t
   // Conclusion: (or (ite C F1 F2) (not F1) (not F2))
   CNF_ITE_NEG3,
 
+  // ========= SAT Refutation for assumption-based unsat cores
+  // Children: (P1, ..., Pn)
+  // Arguments: none
+  // ---------------------
+  // Conclusion: false
+  // Note: P1, ..., Pn corresponds to the unsat core determined by the SAT
+  // solver.
+  SAT_REFUTATION,
```

Comment:
Rather than this rule we can just use a trust rule. There will be no such checking anyway when we are just producing unsat cores.

Reply:
I disagree, TRUST should never be used, it should always be a specific reason, albeit if it is just an identifier.

Notice that this is the reason TRUST is not on master.

Reply:
My recommendation: this rule should exist. The documentation should indicate that this is a "trusted" rule.  See e.g. https://github.com/CVC4/CVC4/blob/master/src/expr/proof_rule.h#L216.

Reply:
I had not noticed that it was not. :) That's fine then.